packages/react-sdk/docusaurus/docs/React/02-tutorials/02-audio-room.mdx

```diff
@@ -3,4 +3,542 @@ title: Audio Room Tutorial
+The entire Stream team loves Typescript. If you're wondering why, consider checking [this post](https://getstream.io/blog/typescript-basics/) about the topic on our Blog.
+:::
+
+Next, we install the dependencies for the Stream package:
+
+```bash
+yarn add @stream-io/video-react-sdk
+```
+
+After that, we can create a new project in the [Stream Dashboard](https://getstream.io/dashboard/). Once logged in, you will be introduced to our dashboard. There you can manage your applications. The app represents Stream's dedicated infrastructure that will handle your app's video calls.
+Make sure to [create a video application](TODO: image/gif showing the UI creating the app) in your Stream dashboard. The API key and secret will be generated automatically upon the app's creation. Those are the credentials we will need for authentication.
+
+:::tip
+If you don't have a Stream account yet, you can register and try it for free [here](https://getstream.io/try-for-free/).
+:::
+
+The last setup step is to create a `.env` file. This is where we will put our API key to keep it safe.
+Create a `.env` file in the root of your project and paste this code in (and replace `<your_api_key>` with the real one):
+
+```
+VITE_STREAM_API_KEY=<your_api_key>
+```
+
+With that the basic setup of the project and the Stream SDK is ready and we can dive into the code.
+
+### User Authentication
+
+We won't cover the entire flow of authentication in this guide because we want to focus on the important parts.
+You can check the complete flow in the sample app repository, and we'll go through the important API calls now.
+
+The first important step is to initialize the Stream client. Therefore, we need the API key and a token provider.
+
+:::tip
+Want to learn why we need the token provider and how it works? See [this page](../../guides/client-auth/#token-providers) for more detail.
+:::
+
+First, we can collect the API key from our `.env` file by simply importing it like this:
+
+```
+const apiKey = import.meta.env.VITE_STREAM_API_KEY as string;
+```
+
+Next, to get a token, we need to create one in our backend. Whenever this expires the SDK will call the token provider again to renew the token.
+There are many different ways to solve this. In this example, we create (and - of course - memoize) a function that uses a `user` object in the same component.
+
+```tsx
+const tokenProvider = useCallback(async () => {
+  const { token } = await fetch(
+    '/api/auth/create-token?' +
+      new URLSearchParams({
+        api_key: apiKey,
+        user_id: user.id,
+      }),
+    {},
+  ).then((res) => res.json());
+  return token as string;
+}, [apiKey, user.id]);
+```
+
+Having this we can create a client object:
+
+```tsx
+const client = useCreateStreamVideoClient({
+  apiKey,
+  tokenOrProvider: tokenProvider,
+  user,
+});
+```
+
+Whenever a user hits the login button we can authenticate them using a call to `connectUser`:
+
+```tsx
+await client.connectUser(user, user.token);
+```
```

Comment:
With the current state, `useCreateStreamVideoClient()` automatically connects the provided user. If you want to delay it, you'll need to use the other, more imperative way of constructing a `StreamVideoClient`:

```ts
const [client] = useState(() => new StreamVideoClient(apiKey));
<StreamVideo client={client}>
   <MyComponent onClick={() => client.connectUser(user)} />
</StreamVideo>
```